docs/v3tov4.md

```diff
@@ -0,0 +1,340 @@
+
+### Entities
+
+In previous versions of the SDK, we had a distinction between the Local Entity and Remote entities. You could get the 
+`Local Entity` calling the method
+
+`func (i *Integration) LocalEntity() *Entity`
+
+or create a `Remote Entity` calling
+
+`func (i *Integration) Entity(name, namespace string, idAttributes ...IDAttribute) (e *Entity, err error)`
+
+In SDK v4, you can obtain the Host Entity or create Entities. The Host Entity is a the entity attached to the host 
+where the agent is running. It inherits the same entity ID as the agent which executes the integration. On the other
+hand, an Entity has its own entity ID. Both types of entities are decorated with the metadata from the host where the
+agent is running.
+
+To get the Host Entity you need to access the `HostEntity` of an integration. Example:
+
+```
+// create the integration
+i, err := integration.New(integrationName, integrationVersion)
+
+// add some inventory to the Host Entity
+i.HostEntity.AddInventoryItem("key", "value", "foo")
+```
+
+To create an Entity you need to call the following method inside  integration
+
+`func (i *Integration) NewEntity(name, type, displayName string) (e *Entity, err error)`
+
+Example: 
+
+```
+// create the integration
+i, err := integration.New(integrationName, integrationVersion)
+
+// create an entity
+entity, err := i.NewEntity(entityName, entityType, entityDisplayName)
+```
+
+These are the parameters accepted by `NewEntity`:
+
+* `entityName` must be an unique value per customer account because it uniquely identifies your Entity. It cannot be empty and 
+  the SDK cannot validate nor enforce that uniqueness, so it's up to the client to define a naming schema that produces 
+  unique names.
+
+  For example, using the following schema `<service_name>:<ip|hostname>:<port>`, we could create these names: 
+  `redis:192.168.100.200:12345` or `mysql:my_server_hostname:8086`
+
+  For the ip or the hostname, please avoid using the values `127.0.0.1` or `localhost` as they don't produce unique 
+  identifiers.
+
+* `entityType` describes the Entity's category. It cannot be empty. Some examples: "RedisInstance", "DockerContainer", 
+  "K8SPod", "CassandraCluster", "MySQLInstance", "KafkaBroker", "NginxServer", etc.
+
+* `entityDisplayName` is a friendly human readable name to be used in the UI. It can be empty.
+
+#### Migrating from LocalEntity
+
+The LocalEntity must be replaced by the Host Entity. The Host Entity has the same methods and functionality as a normal
+Entity. 
+
+#### Adding metadata and tags to an Entity
+
+Entities can have a list of metadata. This metadata can be used to search and filter your entities and will be added to 
+all the metrics and events of the Entity.
+
+There are two types of metadata:
+* Generic metadata about the entity. The metadata is provided by the integration. 
+* Tags. This information is provided by the user through the agent when it executes the integration. The SDK will prefix
```

Comment:
Are tags only provided by the user? All other attributes added to metadata section are not going to be prefixed by `tag.`?

Reply:
Not sure if it's clear what the difference between metadata and tags is.